.github/workflows/lint.yml

```diff
@@ -23,52 +23,75 @@ on:
       - 'pyproject.toml'
       - 'requirements*.txt'
       - 'tox.ini'
-      - 'scripts/*.sh'
-      - '.github/**'
+      - 'scripts/*.sh' # Used by this workflow
+      - '.github/workflows/lint.yml' # This workflow
 
 env:
-  PYTHON_VERSION: 3.11
+  LC_ALL: en_US.UTF-8
+
+defaults:
+  run:
+    shell: bash
+
+permissions:
+  contents: read
 
 jobs:
   lint:
     runs-on: ubuntu-latest
+    name: "${{ matrix.lint.name }}"
+    strategy:
+      fail-fast: false
+      matrix:
+        lint:
+          - name: "ruff"
+            commands: |
+              tox -e ruff -- check
+          - name: "pylint"
+            commands: |
+              echo "::add-matcher::.github/workflows/matchers/pylint.json"
+              tox -e lint
+          - name: "mypy"
+            commands: |
+              tox -e mypy
     steps:
-      - name: Checkout
-        uses: actions/checkout@v4
+      - name: "Harden Runner"
+        uses: step-security/harden-runner@5c7944e73c4c2a096b17a9cb74d65b6c2bbafbde # v2.9.1
+        with:
+          egress-policy: audit # TODO: change to 'egress-policy: block' after couple of runs
+
+      - name: "Checkout"
+        uses: actions/checkout@692973e3d937129bcbf40652eb9f2f61becf3332 # v4.1.7
         with:
           # https://github.com/actions/checkout/issues/249
           fetch-depth: 0
-          submodules: true
 
       - name: Free disk space
         uses: ./.github/actions/free-disk-space
 
       - name: Setup Python 3.11
-        uses: actions/setup-python@v5
+        uses: actions/setup-python@39cd14951b08e74b54015e9e001cdefcf80e669f # v5.1.1
         with:
           python-version: 3.11
           cache: pip
           cache-dependency-path: |
             **/pyproject.toml
             **/requirements*.txt
 
-      - name: Install dependencies
-        id: deps
+      - name: Install tox
         run: |
           python -m pip install --upgrade pip
-          python -m pip install tox
+          python -m pip install tox tox-gh
 
-      - name: Run Ruff check
+      - name: "${{ matrix.lint.name }}"
         run: |
-          tox -e ruff -- check
+          ${{ matrix.lint.commands }}
+        env:
+          RUFF_OUTPUT_FORMAT: github
 
-      - name: Run linting
-        if: ${{ !cancelled() && (steps.deps.outcome == 'success') }}
-        run: |
-          echo "::add-matcher::.github/workflows/matchers/pylint.json"
-          tox -e lint
-
-      - name: Run mypy type checks
-        if: ${{ !cancelled() && (steps.deps.outcome == 'success') }}
-        run: |
-          tox -e mypy
+  lint-workflow-complete:
```

Comment:
Can this be done without having it show up on the jobs list?

Reply:
As far as I'm aware no - that's how it's done in the CLI repo.

A little context as to why - the new matrix scheme splits the linter job definition into three jobs, the motivation for that being we can isolate different facets of the linting (`mypy`, `ruff`, `pylint`) - however, we want mergify to be able to identify whether or not _all_ the linting jobs have passed as a prereq to an automerge.

I think it was @russellb that implemented this scheme originally (please correct me if I'm wrong) - we can explore a diff solution, but my reasoning here was it was best to start from the same point as the CLI repo since we know this works.

Reply:
Thanks for the context!